### ISSUE TITLE
`source` and `.` command do not load environment variables from batchfile which contains `exit /b 1`

### ISSUE THREAD

**Issue by zetamatta** [zetamatta/nyagos#111]
```
<MARKEDONE:~>[1]
$ type ng.bat
set "PROMPT=%PROMPT%+1"
exit /b 1
<MARKEDONE:~>[1]
$ type ok.bat
set "PROMPT=%PROMPT%+1"
exit /b
<MARKEDONE:~>[1]
$ . ng.bat
exit status 1
<MARKEDONE:~>[1]
$ . ok.bat
<MARKEDONE:~>[1]
$ +1
```

**Comment by zetamatta** [comment]
Report found:
- [`make.bat` does not works as expected.](https://twitter.com/h0k0r0bi/status/694144005756456960)

**Comment by zetamatta** [comment]
This issue will be closed because the first problem is fixed on 6060561  and there are not enough data to show the second problem.
